commcare-cloud/commcare_cloud/environment/schemas/fab_settings.py

```diff
@@ -0,0 +1,24 @@
+import jsonobject
+
+GitUriProperty = jsonobject.StringProperty
+TimezoneProperty = jsonobject.StringProperty
+
+class FabSettingsConfig(jsonobject.JsonObject):
+    _allow_dynamic_properties = False
+    sudo_user = jsonobject.StringProperty()
+    default_branch = jsonobject.StringProperty()
+    home = jsonobject.StringProperty()
+    project = jsonobject.StringProperty()
+    code_repo = GitUriProperty()
+    timing_log = jsonobject.StringProperty()
+    keepalive = jsonobject.IntegerProperty()
+    ignore_kafka_checkpoint_warning = jsonobject.BooleanProperty()
+    acceptable_maintenance_window = jsonobject.ObjectProperty(lambda: AcceptableMaintenanceWindow)
```

Comment:
what's the difference between this and `ObjectProperty(AcceptableMaintenanceWindow)`? I've seen both in use.

Reply:
This lets you define `AcceptableMaintenanceWindow` below rather than above this class. (At the time this class is being created `AcceptableMaintenanceWindow`, defined below, does not yet exist. `jsonobject` allows you to use this convention to work around that, emulating Django models, I believe.)

Reply:
👍